### PR TITLE
[bugfix] Do not require a configuration to be resolved always at system level

### LINKED ISSUE #1453: Default environment is required
At the moment, reframe seems to require a default environment. If I do something like this:

```
        {
            'name':'openfoam',
            'target_systems': ['mysys:ib-gcc7-openmpi4-ucx'],
            'modules': ['openfoam-org/7-2ceqb4l']
        },
        {
            'name':'openfoam',
            'target_systems': ['mysys:ib-gcc9-openmpi4-ucx'],
            'modules': ['openfoam-org/7-4zgjbg2']
        },
```

Then I get a message like "environment 'openfoam' not defined for system 'mysys'" (paraphrased, lost the original terminal).

I have to add an empty default environment of the same name for it to work:
```
        {
            'name':'openfoam'
        },
```

Which is fine, because I can use the 'environs' values in the partition to restrict tests to valid partition+environ combinations. *Except* that it is an error trap - for example if I misspelt the 'target_systems' above, then reframe thinks there is a valid combination because of the default one, then I get a cryptic error during module loads.

To me that case with no default seems quite reasonable/normal, but maybe it's not.

### PR DESCRIPTION
**Pull request by vkarak**
Fixes #1453.

### CODE REVIEW

File: reframe/core/systems.py [hunk 1]
```diff
@@ -319,7 +319,7 @@ def create(cls, site_config):
             )
 
         # Restore configuration
-        site_config.select_subconfig(config_save)
+        site_config.select_subconfig(config_save, ignore_resolve_errors=True)
```
Comment (vkarak) [inline reframe/core/systems.py:322]: I think I should add a comment here, why we do that.

File: reframe/frontend/cli.py [hunk 2]
```diff
@@ -463,7 +463,8 @@ def main():
             site_config = config.load_config(converted)
 
         site_config.validate()
-        site_config.select_subconfig(options.system)
+        site_config.select_subconfig(options.system,
+                                     ignore_resolve_errors=True)
```
Comment (vkarak) [inline reframe/frontend/cli.py:467]: I think I should add a comment here, why we do that.

**Comment by vkarak** [comment]
@jenkins-cscs retry daint

**Review by teojgo — APPROVED** [review]
Other than the comments that you have posted I have tested and it looks good.

**Pull request merged**